src/ruby.rs

```diff
@@ -44,7 +44,157 @@ impl_from_arity!(from_arity_13, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE,
 impl_from_arity!(from_arity_14, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE);
 impl_from_arity!(from_arity_15, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE);
 
+/// Data types used within the Ruby C interpreter
+///
+/// * `T_OBJECT` - ordinary Ruby object
+/// * `T_CLASS` - [`Class`](rb_cClass)
+/// * `T_MODULE` - [`Module`](rb_cModule)
+/// * `T_FLOAT` - [`Float`](rb_cFloat)
+/// * `T_STRING` - [`String`](rb_cString)
+/// * `T_REGEXP` - regular expression ([`Regexp`](rb_cRegexp))
+/// * `T_ARRAY` - [`Array`](rb_cArray)
+/// * `T_HASH` - [`Hash`](rb_cHash)
+/// * `T_STRUCT` - [`Struct`](rb_cStruct)
+/// * `T_BIGNUM` - `Bignum`
+/// * `T_FILE` - [`IO`](rb_cIO)
+/// * `T_DATA`
+/// * `T_MATCH` - [`MatchData`](rb_cMatch) object
+/// * `T_COMPLEX` - complex number ([`Complex`](rb_cComplex))
+/// * `T_RATIONAL` - rational number ([`Rational`](rb_cRational))
+/// * `T_NIL` - `nil`
+/// * `T_TRUE` - `true`
+/// * `T_FALSE` - `false`
+/// * `T_SYMBOL` - [`Symbol`](rb_cSymbol)
+/// * `T_FIXNUM` - `Fixnum` (31-bit or 63-bit integer)
+///
+/// In addition, there are several other types used internally:
+///
+/// * `T_NONE`
+/// * `T_IMEMO`
+/// * `T_NODE` - syntax tree node
+/// * `T_ICLASS` - included module
+/// * `T_UNDEF` - undefined
+/// * `T_ZOMBIE` - object awaiting finalization
+/// * `T_MASK`
+///
+/// # Ruby Documentation
+///
+/// * [2.5](https://ruby-doc.org/core-2.5.1/doc/extension_rdoc.html#label-Data+Types)
+#[repr(transparent)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[allow(non_camel_case_types)]
+pub struct ruby_value_type(c_int);
+
 extern {
+    #[doc(hidden)]
```

Comment:
same as 
https://github.com/tildeio/libcruby-sys/pull/61#pullrequestreview-155194251

Reply:
I suspect there _may_ be more to say about each type that could eventually warrant more than a single line of documentation. So probably best to attach the docs ("This type denotes an ordinary Ruby object") to the static, then cross link to it from the bullet point list above. That way we can also generate definition links for each variant.